docs/subnets/walkthrough-prompting.md

```diff
@@ -0,0 +1,117 @@
+
+1. The subnet validator sends a **challenge** simultaneously to multiple subnet miners. This step constitutes the **prompt** from the subnet validator to subnet miners. A challenge is a prompt sent to subnet miners in such a way that:
+   - This prompt is in a style and tone that is similar to humans.
+   - This prompt drives the conversation between the subnet validator and the subnet miners in order to reach a pre-defined goal. 
+
+    See the below [Challenge generation](#challenge-generation) section for how a challenge is generated. 
+2. The subnet miners respond to the subnet validator after performing the challenge **task**. 
+3. The subnet validator then **scores** each subnet miner by comparing the subnet miner's response with a locally generated **reference** answer. The subnet validator uses this reference as the ground truth for this step. 
+4. Finally, the subnet validator **sets the weights** for the subnet miners by sending the weights to the blockchain. In the Bittensor blockchain the Yuma Consensus allocates the rewards to the participating subnet miners and subnet validators. 
+
+:::tip Use of large language models 
+In this subnet both the subnet validator and the subnet miners use large language models (LLMs) to create the challenges (subnet validator) and respond to the prompts (subnet miners).
+:::
+
+## Key innovations in this subnet
+
+This subnet has developed a few innovative techniques to get to a real human-like conversational AI that actually produces intelligence instead of copying from the internet. See below:
+
+### Achieving human-like conversation
+
+To deliver to a user of this subnet an experience of a human-like conversation:
+
+- Subnet validators perform a roleplay where they take on the persona of **random** human users before they prompt the subnet miners. By doing this, the subnet validators engage the subnet miners in a real, random, human-like conversation throughout the subnet operation.
+- This means that throughout the subnet validation process the subnet miners continually improve their performance as though all validation prompts originated from an external API client query. This enables the subnet miners to become better intelligent AI assistants. 
+
+:::tip Class HumanAgent
+See [class HumanAgent](https://github.com/opentensor/prompting/blob/main/prompting/agent.py#L30).
+:::
+
+### Prevent subnet miners from looking up the answers
+
+To prevent the subnet miners from simply looking up the answers on the internet, this subnet incorporates a novel approach&mdash;it introduces fuzziness into the prompt and requires that the subnet miners use semantic intelligence to understand the instructions contained in the prompt.
+
+### Evolve subnet as a mixture of experts (MoE)
+
+The subnet validator composes a challenge based on whether the task is answering questions, summarizing a given text, debugging code, solve mathematics problems. The motivation behind using multiple tasks is several fold:
```

Comment:
```suggestion
The subnet validator composes a challenge based on whether the task is answering questions, summarizing a given text, debugging code, solve mathematics problems and so on. The motivation behind using multiple tasks is several fold:
```

Reply:
Let’s keep it open ended. We add new tasks all the time to mimic the distribution of real use cases and benchmark the miner performance across a broad skill set (which is also useful for a routing mechanism)